File: bbarena-model/src/main/java/net/sf/bbarena/model/Weather.java
package net.sf.bbarena.model;

public interface Weather {

    enum WeatherType {
        SWELTERING_HEAT,
        VERY_SUNNY,
        NICE,
        POURING_RAIN,
        BLIZZARD
    }

    static Weather getWeather(WeatherType weatherType) {
        return getWeather(weatherType.name());
    }

    static Weather getWeather(String weatherType) {
        return new Weather() {
            @Override
            public String getName() {
                return weatherType;
            }
            @Override
            public String toString() {
                return getName();
            }
        };
    }

    String getName();

}

File: src/main/kotlin/bbarena/model/Weather.kt
package bbarena.model

fun getWeather(weatherType: WeatherType): Weather = getWeather(weatherType.name)

fun getWeather(weatherType: String): Weather = Weather(weatherType)

enum class WeatherType {
    SWELTERING_HEAT,
    VERY_SUNNY,
    NICE,
    POURING_RAIN,
    BLIZZARD
}

data class Weather(val name: String = WeatherType.NICE.name)
